scalameta/tokenizers/shared/src/main/scala/scala/meta/internal/tokenizers/LegacyScanner.scala

```diff
@@ -536,30 +536,20 @@ class LegacyScanner(input: Input, dialect: Dialect) {
   }
 
   @tailrec
-  private def getIdentRest(): Unit = (ch: @switch) match {
-    case 'A' | 'B' | 'C' | 'D' | 'E' | 'F' | 'G' | 'H' | 'I' | 'J' | 'K' | 'L' | 'M' | 'N' | 'O' |
-        'P' | 'Q' | 'R' | 'S' | 'T' | 'U' | 'V' | 'W' | 'X' | 'Y' | 'Z' | '$' | 'a' | 'b' | 'c' |
-        'd' | 'e' | 'f' | 'g' | 'h' | 'i' | 'j' | 'k' | 'l' | 'm' | 'n' | 'o' | 'p' | 'q' | 'r' |
-        's' | 't' | 'u' | 'v' | 'w' | 'x' | 'y' | 'z' | '0' | '1' | '2' | '3' | '4' | '5' | '6' |
-        '7' | '8' | '9' =>
-      if (ch == '$' && !getDollar()) { finishNamed(); return }
+  private def getIdentRest(): Unit = {
+    if (ch == '_') {
       putChar(ch)
       nextChar()
-      getIdentRest()
-    case '_' =>
+      if (isIdentifierPart(ch))
+        getIdentRest()
+      else
+        getOperatorRest()
+    } else if ( // strangely enough, Character.isUnicodeIdentifierPart(SU) returns true!
+      ch != SU && (if (ch == '$') getDollar() else Character.isUnicodeIdentifierPart(ch))) {
```

Comment:
Could you extract the condition after `&&` to a def? It seems a bit hard to understand currently.

Reply:
would it be easier to read if i change it to a match? SU vs $ vs rest?

Reply:
Might be, but why not change it to a def ?

Reply:
mostly because i don't know what to call this `def` :) any suggestions?